### PR TITLE
Added --disable-proxy

### PR DESCRIPTION
**Pull request by Anumita**
---
Version increase in setup.py will be taken up in another PR

Added --disable-proxy to go from a proxy scenario to a non proxy scenario
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/connectedk8s/azext_connectedk8s/_params.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def load_arguments(self, _):
         c.argument('http_proxy', options_list=['--proxy-http'], help='Http proxy URL to be used.')
         c.argument('no_proxy', options_list=['--proxy-skip-range'], help='List of URLs/CIDRs for which proxy should not to be used.')
         c.argument('proxy_cert', options_list=['--proxy-cert'], type=file_type, completer=FilesCompleter(), help='Path to the certificate file for proxy')
+        c.argument('disable_proxy', options_list=['--disable-proxy'], action='store_true', help='Disables applying proxy to agents')
```
Comment (shashankbarsin) [inline src/connectedk8s/azext_connectedk8s/_params.py:36]:
```suggestion
        c.argument('disable_proxy', options_list=['--disable-proxy'], action='store_true', help='Disables proxy settings for agents')
```

**Review by shashankbarsin — CHANGES_REQUESTED** [review]

File: src/connectedk8s/azext_connectedk8s/_constants.py [hunk 2]
```diff
@@ -45,3 +45,4 @@
 Kubeconfig_Failed_To_Load_Fault_Type = "failed-to-load-kubeconfig-file"
 Proxy_Cert_Path_Does_Not_Exist_Fault_Type = 'proxy-cert-path-does-not-exist-error'
 Proxy_Cert_Path_Does_Not_Exist_Error = 'Proxy cert path {} does not exist. Please check the path provided'
+Nothing_To_Update_Error = 'No parameters found to update. Please run az connectedk8s update --help to see which parameter you want to update'
```
Comment (shashankbarsin) [inline src/connectedk8s/azext_connectedk8s/_constants.py:48]:
```suggestion
Nothing_To_Update_Error = "No parmeters were specified with update command. Please run az connectedk8s update --help to check parameters available for update'
```

**Review by shashankbarsin — CHANGES_REQUESTED** [review]

File: src/connectedk8s/azext_connectedk8s/_constants.py (hunk 2 above)
Comment (Anumita) [reply src/connectedk8s/azext_connectedk8s/_constants.py]: done

File: src/connectedk8s/azext_connectedk8s/_params.py (hunk 1 above)
Comment (Anumita) [reply src/connectedk8s/azext_connectedk8s/_params.py]: done

File: src/connectedk8s/azext_connectedk8s/_constants.py [hunk 3]
```diff
@@ -45,3 +45,4 @@
 Kubeconfig_Failed_To_Load_Fault_Type = "failed-to-load-kubeconfig-file"
 Proxy_Cert_Path_Does_Not_Exist_Fault_Type = 'proxy-cert-path-does-not-exist-error'
 Proxy_Cert_Path_Does_Not_Exist_Error = 'Proxy cert path {} does not exist. Please check the path provided'
+Nothing_To_Update_Error = 'No parmeters were specified with update command. Please run az connectedk8s update --help to check parameters available for update'
```
Comment (krdhruva) [inline src/connectedk8s/azext_connectedk8s/_constants.py:48]: No_Param_Error :)

**Comment by yonzhan** [comment]
connectedk8s

File: src/connectedk8s/azext_connectedk8s/_constants.py (hunk 3 above)
Comment (Anumita) [reply src/connectedk8s/azext_connectedk8s/_constants.py]: done

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 4]
```diff
@@ -837,6 +842,10 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
         cmd_helm_upgrade.extend(["--set", "global.httpProxy={}".format(http_proxy)])
     if no_proxy:
         cmd_helm_upgrade.extend(["--set", "global.noProxy={}".format(no_proxy)])
+    if https_proxy or http_proxy or no_proxy:
+        cmd_helm_upgrade.extend(["--set", "global.isProxyEnabled={}".format(True)])
+    if disable_proxy:
+        cmd_helm_upgrade.extend(["--set", "global.isProxyEnabled={}".format(False)])
```
Comment (krdhruva) [inline src/connectedk8s/azext_connectedk8s/custom.py:848]: I think we should handle the case where customer by mistake pass proxy params as well as --disable proxy.

Comment (Anumita) [reply src/connectedk8s/azext_connectedk8s/custom.py:851]: done

**Review by shashankbarsin — APPROVED** [review]

**Review by krdhruva — APPROVED** [review]

**Comment by Anumita** [comment]
@fengzhou-msft , review is done from the connectedk8s side. It is ready for checking in if you don't have any further reviews

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**